packages/core/src/services/health/types.ts

```diff
@@ -96,30 +86,51 @@ export interface IHealthResponseDetails {
   secondsToRecover?: number;
 }
 
-export interface IReferenceURL {
-  url: string;
-  auth?: string;
+export interface IHealthResponseParams {
+  name: string;
+  result?: any;
+  height?: IBlockHeight;
+  secondsToRecover?: number;
+  badOracles?: string[];
+  noOracle?: boolean;
+  error?: Error;
 }
 
-export interface IRPCResponse {
-  jsonrpc: string;
-  id: number;
-  result?: string | number;
-  error?: { code: number; message: string };
+export interface INodeCheckParams {
+  host: string;
+  port: number;
 }
 
-export interface IRPCSyncResponse {
-  jsonrpc: string;
-  id: number;
-  result: boolean;
-  error?: { code: number; message: string };
+export interface IOraclesResponse {
+  oracleHeights: number[];
+  badOracles: string[];
 }
 
-export interface IEVMHealthCheckOptions {
-  harmony?: boolean;
+export interface IRefHeight {
+  refHeight: number;
+  badOracles?: string[];
+  noOracle?: boolean;
 }
 
-export interface IOraclesResponse {
-  healthyOracles: IReferenceURL[];
-  badOracles: IReferenceURL[];
+export interface IRPCCheckParams {
+  chain: IChain;
+  url: string;
+  basicAuth?: string;
+}
+
+export interface IRPCMethodParams {
+  fullRpcUrl: string;
+  basicAuth?: string;
+  rpc?: { jsonrpc: string; id: number; method: string; params?: any[] };
+}
+
+export interface IRPCResult {
+  jsonrpc: string;
+  id: number;
+  result?:
+    | string // ALG, SOL
+    | number
+    | { healthy?: boolean } // AVAX
```

Comment:
do these chain-specific types mean we would still require code-level updates when adding new chains? anyway we can not have to do that, generate dynamically or skip?

Reply:
It's just Typescript types so not really important as its only compile time. 

Can easily make it `any` but was just using those as examples while converting what we had there as those are the possible types for our existing chains; the types themselves may well end up being some that are not in that list so maybe have it be `any` is best since we may not know all of that that field looks like.

Reply:
Changed to `any`